geojson/src/commonMain/kotlin/org/maplibre/spatialk/geojson/Feature.kt

```diff
@@ -60,20 +60,21 @@ public class Feature(
 
     override fun json(): String = GeoJson.encodeToString(this)
 
-    public fun copy(
+    @Suppress("UNCHECKED_CAST")
+    public fun <T : Geometry> xcopy(
         geometry: Geometry? = this.geometry,
         properties: JsonObject? = this.properties,
         id: String? = this.id,
         bbox: BoundingBox? = this.bbox,
-    ): Feature = Feature(geometry, properties, id, bbox)
+    ): Feature<T> = Feature(geometry as T, properties, id, bbox)
```

Comment:
ah damn, so `out T` on the class means we can't accept a `geometry: T?` here, but we need `out T` for a `foo: Feature<Geometry>` to accept a `Feature<Point>`?

I don't like the unchecked case here, but at least #201 will fix it by using the kotlin generated `.copy()` which seems to handle this case fine somehow

Reply:
merged main and resolved